libsolidity/parsing/Scanner.cpp

```diff
@@ -753,10 +773,22 @@ Token::Value Scanner::scanNumber(char _charSeen)
 				// hex number
 				kind = HEX;
 				addLiteralCharAndAdvance();
+				skipUnderscore();
 				if (!isHexDigit(m_char))
 					return Token::Illegal; // we must have at least one hex digit after 'x'/'X'
-				while (isHexDigit(m_char))
+				while (isHexDigit(m_char) || m_char == '_') // same logic as scanDecimalDigits
```

Comment:
This doesn't prevent the leading underscore which seems to be weird in hex (`0x_1234`)

Reply:
I added that as python allows it:
 (an example from [https://www.python.org/dev/peps/pep-0515/](https://www.python.org/dev/peps/pep-0515/))
flags = 0b_0011_1111_0100_1110